hack/update-library-go.sh

```diff
@@ -0,0 +1,125 @@
+#!/bin/bash
+
+# A tool to rebuild and update kube-controller-manager-operator and
+# kube-apiserver-operator in a running cluster against changes in a local
+# library-go repo.
+#
+# Usage: update-library-go.sh <library-go path> <operator source directory>
+#
+# <library-go path> is a local directory library-go
+#
+# <operator source directory> is a local directory where operators will be
+# checked out if they don't exist already. The directory will be created if it
+# doesn't exist.
+
+set -e -o pipefail
+
+# Ensure the cluster registry's default route is enabled, and return the route
+function get_image_registry {
+    local url
+
+    oc patch configs.imageregistry.operator.openshift.io/cluster --type=merge \
+        --patch '{"spec":{"defaultRoute":true}}' >/dev/null
+
+    while ! url=$(oc get -n openshift-image-registry route/default-route -o json | jq -re .spec.host); do
+        echo Waiting for default route
+        sleep 1
+    done
+
+    echo $url
+}
+
+# Login to the cluster's internal registry using a token
+function registry_login {
+    local url=$1
+    local token=$2
+
+    podman login --tls-verify=false -u kubeadmin -p "${token}" $url
+}
+
+# Rebuild the operator image, push it to the cluster's internal registry, and
+# update the operator deployment to use the new image
+function update_operator {
+    local name=$1
+
+    local repo="cluster-${name}"
+    local namespace="openshift-${name}"
+    local image="${namespace}/${name}"
+    local sourcedir="${workingdir}/${name}"
+
+    [ ! -d "${sourcedir}" ] && \
+        git clone "https://github.com/openshift/${repo}.git" "${sourcedir}"
+    pushd "${sourcedir}"
+        go mod edit --replace "github.com/openshift/library-go=$librarygo"
+        go mod vendor
+
+        # This dance is so we get build output during execution while also
+        # capturing it
+        mkfifo /tmp/buildah.$$
+        buildah bud -t "${image}" Dockerfile.rhel7 | tee /tmp/buildah.$$ &
```

Comment:
Would prefer having a way to use podman/docker for those tasks.

Reply:
buildah is from the same stable as podman. It's the the bit that does builds. `podman build` will use `buildah bud` under the covers.

We could add docker support, but given that AFAIK we don't support it I'm not sure it's worth the hassle.